src/classes/User.js

```diff
@@ -33,63 +33,72 @@ class User {
   removeFromRecipesToCook(recipeOut) {
     this.recipesToCook.forEach(recipe => {
       if (recipeOut.name === recipe.name) {
-        this.recipesToCook.splice(recipeOut, 1)
+        this.recipesToCook.splice(this.recipesToCook.indexOf(recipeOut), 1)
       }
     })
   }
 
-  filterFavoriteRecipeTags(tags) {
-    let matchTags = [];
-    this.favoriteRecipes.forEach(recipe => {
-      tags.forEach(tag => {
-        if (recipe.tags.includes(tag)) {
-          matchTags.push(recipe);
-        }
-      })
-    })
-    if (!matchTags.length) {
-      return "No recipes match criteria."
-    } else {
-      return matchTags
-    }
+  filterFavoriteRecipeTags(inputTag) {
+    let tagged = this.favoriteRecipes.filter(recipe => recipe.tags.includes(inputTag));
+    return tagged;
   }
 
-  filterFavoriteRecipeByName(name) {
```

Comment:
These are simple adjustments to use the same term for the search input, across all classes